docs/docs/reference/sdk/search/Overview.md

```diff
@@ -0,0 +1,61 @@
+
+![Faceted Search](/img/references/faceted-search.png)
+
+## How the `Search` module works?
+
+The `Search` SDK module provides a standard for creating and modifying **faceted search** states. Faceted navigation generates a unique and serialized URL for every facet combination, which we'll call facet states. 
+Developers can store facet states in the browser's [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API) to navigate through the user's history and manipulate a website's URL without a full page refresh. This way, one can create a rich, router-independent way for applying and removing facets.
+
+Three main files shape the behavior of the `Search` SDK module:
+- [`useSearchState`](https://github.com/vtex/faststore/blob/master/packages/sdk/src/search/useSearchState.ts)
+- [`serializer`](https://github.com/vtex/faststore/blob/master/packages/sdk/src/search/serializer.ts)
+- [`useInfiniteSearchState`](https://github.com/vtex/faststore/blob/master/packages/sdk/src/search/useInfiniteSearchState.ts)
+  
+### useSearchState
+
+The `useSearchState` file defines the interfaces that describe a faceted search state and the functions that change these states.
+
+The main `State` interface allows you to handle the following variables:
+
+|Argument|Type        |Description|
+|--------|------------|---------------------------------------------------|
+|`base`  |`string`    | Base URL path for the search context. This is important when the search context is a subset of what the browser is currently showing to the user, like in an internationalized context where your pages are scoped by a `/locale` param like in `/en` or `/pt`|
+|`page`  |`number`    ||
+|`selectedFacets`|`Facet[]`|All selected facets.|
+|`sort`  | `SearchSort`|Selected sorting criteria for the search (e.g, ascending/descending price, number of reviews)|
+|`term`  |`string`    | Terms of the full text search.|
+
+To change the facet state, the following functions are also available:
+- `setFacet` - Select a facet by its key.
+- `toggleFacet` - Replace a selected facet by key.
+- `removeFacet` - Remove the selected facet by its value.
+
+### serializer
+
+The `serializer` file handles URL parsing to create a serialized URL for every facet combination.
+
+### useInfiniteSearchState
+
+The `useInfiniteSearchState` file is an optional function you can use to add infinite scroll in your faceted search pages. The `useInfiniteSearchState` file keeps a record of which pages and products have already been displayed to shoppers with the infinite scroll pagination.
+
+## Example
+
+You can find some examples and boundary conditions [in our tests.](https://github.com/vtex/faststore/tree/master/packages/sdk/test/search)
```

Comment:
I felt a bit lost with this example. We talked about `useSearchState`, `serializer` and `useInfiniteSearchState`, but I'm not sure about how to use them in this example.

Reply:
Also, as an Overview doc, I think we need to talk briefly about what is inside this section:
- SearchProvider
- usePagination
- useSearch

Reply:
Ok. I did some refactoring on this part, tell me what you think about it